src/components/UpdateSubscription/UpdateSubscription.tsx

```diff
@@ -0,0 +1,33 @@
+import { useAppSelector } from 'appRedux/store';
+import Unsubscribe from './Unsubscribe';
+import Resubscribe from './Resubscribe';
+
+import { UpdateSubscriptionProps } from './UpdateSubscription.types';
+
+const UpdateSubscription = ({
+  customCancellationReasons,
+  skipAvailableDowngradesStep,
+  skipAvailableFreeExtensionStep,
+  skipCancellationSurveyStep
+}: UpdateSubscriptionProps) => {
+  const { updateSubscription } = useAppSelector((state) => state.popupManager);
+
+  if (updateSubscription?.action === 'unsubscribe') {
```

Comment:
please refactor this into a `switch` statement. the `default` should rather be `return null;` instead of an empty fragment, not to create a node.